src/DotNetWorker/Hosting/ServiceCollectionExtensions.cs

```diff
@@ -74,33 +53,41 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorker(this IServic
             // Output Bindings
             services.AddSingleton<IOutputBindingsInfoProvider, DefaultOutputBindingsInfoProvider>();
 
-            // gRpc
-            services.AddSingleton<IWorker, GrpcWorker>();
-            services.AddSingleton<FunctionRpcClient>(p =>
-            {
-                IOptions<GrpcWorkerStartupOptions> argumentsOptions = p.GetService<IOptions<GrpcWorkerStartupOptions>>();
-                GrpcWorkerStartupOptions arguments = argumentsOptions.Value;
-
-                GrpcChannel grpcChannel = GrpcChannel.ForAddress($"http://{arguments.Host}:{arguments.Port}", new GrpcChannelOptions()
-                {
-                    Credentials = ChannelCredentials.Insecure
-                });
-
-                return new FunctionRpcClient(grpcChannel);
-            });
+            // Worker initialization service
             services.AddSingleton<IHostedService, WorkerHostedService>();
-            services.AddOptions<GrpcWorkerStartupOptions>()
-                .Configure<IConfiguration>((arguments, config) =>
-                {
-                    config.Bind(arguments);
-                });
 
             if (configure != null)
             {
                 services.Configure(configure);
             }
 
-            return new FunctionsWorkerApplicationBuilder(services);
+            return new FunctionsWorkerApplicationBuilder(services); ;
+        }
+
+        /// <summary>
+        /// Adds the core set of services for the Azure Functions worker.
+        /// This call also adds the default set of binding converters and gRPC support.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        /// <param name="configure">The action used to configure <see cref="WorkerOptions"/>.</param>
+        /// <returns>The same <see cref="IFunctionsWorkerApplicationBuilder"/> for chaining.</returns>
+        public static IFunctionsWorkerApplicationBuilder AddFunctionsWorkerDefaults(this IServiceCollection services, Action<WorkerOptions>? configure = null)
```

Comment:
Could this be something like `AddDefaultFunctionsWorker`? I'm not sure why, but ending it with `Defaults` was hard for me to parse :-)

Reply:
Yeah, at first, it felt  odd, but it's consistent with `ConfigureWebHostDefaults` for ASP.NET, and it does make sense when you think about what it does; it gives you the default services and you can configure them (this is the `Add` version against the service collection)

Reply:
Ah okay -- if this naming exists already, I'm good.